src/MOIWrapper.jl

```diff
@@ -0,0 +1,358 @@
+        push!(zero_one_indices, mapping.varmap[f.variable].value)
+    end
+end
+
+function update_integer_indices(user_optimizer::MOI.ModelLike, mapping::MOIU.IndexMap,
+    ci::Vector{MOI.ConstraintIndex{F,S}}, integer_indices::Vector{Int}) where {F, S}
+    for i in 1:length(ci)
+        f = MOI.get(user_optimizer, MOI.ConstraintFunction(), ci[i])
+        push!(integer_indices, mapping.varmap[f.variable].value)
+    end
+end
+
+"""
+Receive a cbc_optimizer which contains the pointer to the cbc C object and instanciate the object cbc_model_format::CbcModelFormat based on user_optimizer::AbstractModel (also provided by the user).
+Function loadProblem of CbcCInterface requires all information stored in cbc_model_format.
+"""
+function MOI.copy!(cbc_optimizer::CbcOptimizer,
+    user_optimizer::MOI.ModelLike; copynames=false)
+
+    mapping = MOIU.IndexMap()
+
+    num_cols = MOI.get(user_optimizer, MOI.NumberOfVariables())
+    var_index = MOI.get(user_optimizer, MOI.ListOfVariableIndices())
+    for i in 1:num_cols
+        mapping.varmap[var_index[i]] = MOI.VariableIndex(i)
+    end
+
+    zero_one_indices = Int[]
+    integer_indices = Int[]
+    list_of_constraints = MOI.get(user_optimizer, MOI.ListOfConstraints())
+    num_rows = 0
+    for (F,S) in list_of_constraints
+        if !(MOI.supportsconstraint(cbc_optimizer, F, S))
+            return MOI.CopyResult(MOI.CopyUnsupportedConstraint,
+            "Cbc MOI Interface does not support constraints of type " * (F,S) * ".", nothing)
+        end
+
+        ci = MOI.get(user_optimizer, MOI.ListOfConstraintIndices{F,S}())
+
+        if S == MOI.ZeroOne
+            update_zeroone_indices(user_optimizer, mapping, ci, zero_one_indices)
+        elseif S == MOI.Integer
+            update_integer_indices(user_optimizer, mapping, ci, integer_indices)
+        end
+
+        ## Update conmap for (F,S) for F != MOI.SingleVariable
+        ## Single variables are treated by bounds in Cbc, so no
+        ## need to add a row
+        if F != MOI.SingleVariable
+            for i in 1:length(ci)
+                mapping.conmap[ci[i]] = MOI.ConstraintIndex{F,S}(num_rows + i)
+            end
+            num_rows += MOI.get(user_optimizer, MOI.NumberOfConstraints{F,S}())
+        end
+    end
+
+    cbc_model_format = CbcModelFormat(num_rows, num_cols)
+
+    copy_constraints!(cbc_model_format, user_optimizer, mapping)
+    update_bounds_for_binary_vars!(cbc_model_format.col_lb, cbc_model_format.col_ub, zero_one_indices)
+
+
+    ## Copy objective function
+    objF = MOI.get(user_optimizer, MOI.ObjectiveFunction{MOI.ScalarAffineFunction{Float64}}())
+    load_obj(cbc_model_format, mapping, objF)
+    sense = MOI.get(user_optimizer, MOI.ObjectiveSense())
+    MOI.set!(cbc_optimizer, MOI.ObjectiveSense(), sense)
+
+    ## Load the problem to Cbc
+    CbcCI.loadProblem(cbc_optimizer.inner, sparse(cbc_model_format.row_idx, cbc_model_format.col_idx,
```

Comment:
Clear `row_idx`, `col_idx`, and `values` after this call to save memory?

Reply:
Fixed